smarts/core/agent_manager.py

```diff
@@ -319,7 +321,12 @@ def switch_initial_agents(self, agent_interfaces: Dict[str, AgentInterface]):
 
     def setup_agents(self, sim):
         self.init_ego_agents(sim)
-        self.setup_social_agents(sim)
+        if sim.scenario.social_agents:
+            if not self._remote_agent_buffer:
+                self._remote_agent_buffer = RemoteAgentBuffer(
+                    zoo_manager_addrs=self._zoo_addrs
+                )
+            self.setup_social_agents(sim)
```

Comment:
I think the `RemoteAgentBuffer` instance creation should sit in the `setup_social_agents()` to eliminate this private call dependency for the public `setup_social_agents()`.

Reply:
The `RemoteAgentBuffer` class creation has been moved into `setup_social_agents()` function.